pkg/cmd/issue/argparsetest/argparsetest.go

```diff
@@ -0,0 +1,129 @@
+package argparsetest
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/google/shlex"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// newCmdFunc represents the typical function signature we use for creating commands e.g. `NewCmdView`.
+//
+// It is generic over `T` as each command construction has their own Options type e.g. `ViewOptions`
+type newCmdFunc[T any] func(f *cmdutil.Factory, runF func(*T) error) *cobra.Command
+
+func TestArgParsing[T any](t *testing.T, fn newCmdFunc[T]) {
```

Comment:
Looking through the existing tests, I noticed that the coverage was pretty smattered. Nearly all issue commands have: 1 positional argument `number | url`. All those commands now use this for argument validation. Commands with extra flags use this **and** further table tests. `transfer` is unable to use this as it has another required positional argument.